UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs

```diff
@@ -0,0 +1,82 @@
+﻿#nullable enable
+
+using DotNetCheck.DotNet;
+using DotNetCheck.Models;
+using DotNetCheck.Solutions;
+using NuGet.Versioning;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Checkups;
+
+internal abstract class DotNetNewTemplatesCheckupBase : Checkup
+{
+    public abstract string TemplatesDisplayName { get; }
+    
+    public abstract string PackageName { get; }
+
+    public abstract Regex DotNetNewOutputRegex { get; }
+
+    public override async Task<DiagnosticResult> Examine(SharedState history)
+    {
+        var dotnetOutput = GetDotNetNewInstalledList();
+        var version = GetInstalledVersion(dotnetOutput, DotNetNewOutputRegex);
+        if (version is null)
+        {
+            return new(
+                Status.Error,
+                this,
+                new Suggestion(
+                    $"The {TemplatesDisplayName} dotnet new templates are not installed.",
+                    new DotNetNewTemplatesInstallSolution(PackageName)));
+        }
+        
+        var latestVersion = await NuGetHelper.GetLatestPackageVersionAsync(
+            PackageName,
+            ToolInfo.CurrentVersion.IsPrerelease);
+
+        if (latestVersion > version)
+        {
+            return new(
+                Status.Error,
+                this,
+                new Suggestion(
+                    $"The {TemplatesDisplayName} dotnet new templates are not up to date " +
+                    $"(installed version {version}, latest available version {latestVersion}.",
+                    new DotNetNewTemplatesInstallSolution(PackageName, latestVersion)));
+        }
+
+        return DiagnosticResult.Ok(this);
+    }
+
+    private string GetDotNetNewInstalledList()
+    {
+        // Running 'dotnet new uninstall' without any package ID will list all installed
+        // dotnet new templates along with their versions.
+        var processInfo = new ProcessStartInfo("dotnet", "new uninstall");
```

Comment:
A more robust approach would be using `Microsoft.TemplateEngine.Edge` NuGet package. I haven't used it before, but I believe this is what the .NET CLI uses underneath the hoods.

That way we don't have to worry about .NET CLI changing the format of the produced output.

Reply:
Tried to use the library, but I wasn't able to get it work and there is not many useful samples for it. So for now, keeping the original approach, but if you have some resources, I would look into it further 👍